pkg/client/build.go

```diff
@@ -1042,6 +1043,30 @@ func (c *Client) fetchBuildpack(ctx context.Context, bp string, relativeBaseDir
 		fetchedBPs = append(append(fetchedBPs, mainBP), depBPs...)
 		mainBPInfo := mainBP.Descriptor().Info()
 		moduleInfo = &mainBPInfo
+
+		packageCfgPath := filepath.Join(bp, "package.toml")
+		_, err = os.Stat(packageCfgPath)
+		if err == nil {
+			packageReader := buildpackage.NewConfigReader()
+			packageCfg, err := packageReader.Read(packageCfgPath)
+			if err == nil {
+				for _, dep := range packageCfg.Dependencies {
+					mainBP, deps, err := c.buildpackDownloader.Download(ctx, dep.URI, buildpack.DownloadOptions{
+						RegistryName:    registry,
+						ImageOS:         imageOS,
+						RelativeBaseDir: filepath.Join(bp, packageCfg.Buildpack.URI),
+						Daemon:          !publish,
+						PullPolicy:      pullPolicy,
+					})
+
+					if err != nil {
+						return nil, nil, errors.Wrapf(err, "fetching dependencies (uri=%s,image=%s)", style.Symbol(dep.URI), style.Symbol(dep.ImageName))
```

Comment:
Codecov is complaining about the missing test for handling this error, I think would be better to cover it up as well